src/bedrock/kvs/server.cpp

```diff
@@ -446,6 +446,38 @@ void run(unsigned thread_id, Address ip, Address seed_ip,
         push_request(req, pushers[target_address]);
       }
 
+      // report key size stats
+      communication::Key_Size primary_key_size;
+      for (const auto& key_size_pair : key_size_map) {
+        if (is_primary_replica(key_size_pair.first, placement, global_hash_ring_map,
+                        local_hash_ring_map, wt)) {
+          communication::Key_Size_Tuple* tp = primary_key_size.add_tuple();
+          tp->set_key(key_size_pair.first);
+          tp->set_size(key_size_pair.second);
+        }
+      }
+
+      key = std::string(kMetadataIdentifier) + "_" + wt.get_ip() + "_" +
```

Comment:
Since we're dealing with a bunch of metadata here, does it make sense to loop in #82 and clean that up as well? 

For future reference, it's not good practice to resolve multiple, non-trivial issues in a single PR, but I think we're optimizing for expediency here. :-)

Reply:
These are all the issues related to the monitoring side, so I grouped them into one PR. But as it turned out it also required some change to the server end...

Let's make #82 a separate PR.

Reply:
👍